crates/pallets/starknet/src/lib.rs

```diff
@@ -753,32 +725,62 @@ pub mod pallet {
             // Once we have a real fee market this is where we'll chose the most profitable transaction.
 
             let transaction = Self::get_call_transaction(call.clone()).map_err(|_| InvalidTransaction::Call)?;
+            // Important to store the nonce before the call to prevalidate, because the `handle_nonce`
+            // function will increment it
+            let transaction_nonce = get_transaction_nonce(&transaction);
+            let sender_address = get_transaction_sender_address(&transaction);
+            let sender_nonce = Self::nonce(sender_address);
 
-            let tx_priority_info = Self::validate_unsigned_tx_nonce(&transaction)?;
-
-            Self::validate_unsigned_tx(&transaction)?;
+            Self::pre_validate_unsigned_tx(&transaction)?;
 
             let mut valid_transaction_builder = ValidTransaction::with_tag_prefix("starknet")
                 .priority(u64::MAX)
                 .longevity(T::TransactionLongevity::get())
                 .propagate(true);
 
-            match tx_priority_info {
-                // Make sure txs from same account are executed in correct order (nonce based ordering)
-                TxPriorityInfo::RegularTxs { sender_address, transaction_nonce, sender_nonce } => {
+            match &transaction {
+                Transaction::AccountTransaction(_) => {
                     valid_transaction_builder =
-                        valid_transaction_builder.and_provides((sender_address, Felt252Wrapper(transaction_nonce.0)));
-                    if transaction_nonce > sender_nonce {
-                        valid_transaction_builder = valid_transaction_builder
-                            .and_requires((sender_address, Felt252Wrapper(transaction_nonce.0 - FieldElement::ONE)));
-                    }
+                        valid_transaction_builder.and_provides((sender_address, transaction_nonce));
+
+                    match (transaction_nonce, sender_nonce) {
+                        // Special case where the wallet send both deploy_account and first tx at the same time
+                        // The first tx validation would fail because the contract is not deployed yet,
+                        // so we skip the entrypoint execution for now
+                        (Nonce(StarkFelt::ONE), Nonce(StarkFelt::ZERO)) => {
```

Comment:
So I can implement it, I did, but the end result is exactly the same. You run a call to the `__validate__` entrypoint, that can only fail, and you then say "It's okay to fail there". But we already knew it would fail, because the Nonce is 0, meaning there is no way anything is deployed to this address at the time.

Reply:
> with invalid txs of nonce

There is no way to know if the txs with nonce one are valid or not. They, we simply cannot run the `__validate__` entrypoint because it is not deployed. We have to blindly accept them in the mempool.

Reply:
Ya you're right, my bad.